performance_regression_gradual_grow_throughput.py

```diff
@@ -0,0 +1,236 @@
+        params = {}
+        stress_queue = []
+        if self.params.get('round_robin'):
+            self.log.debug('Populating data using round_robin')
+            params.update({'stress_num': 1, 'round_robin': True})
+        if compaction_strategy:
+            self.log.debug('Next compaction strategy will be used %s', compaction_strategy)
+            params['compaction_strategy'] = compaction_strategy
+
+        for stress_cmd in population_commands:
+            params.update({'stress_cmd': stress_cmd})
+            # Run all stress commands
+            params.update(dict(stats_aggregate_cmds=False))
+            self.log.debug('RUNNING stress cmd: {}'.format(stress_cmd))
+            stress_queue.append(self.run_stress_thread(**params))
+
+        for stress in stress_queue:
+            self.get_stress_results(queue=stress, store_results=False)
+
+        self.log.info("Dataset has been populated")
+
+    # pylint: disable=too-many-arguments,too-many-locals
+    def run_gradual_increase_load(self, stress_cmd_templ,
+                                  start_ops, max_ops, throttle_step,
+                                  stress_num, num_loaders, compaction_strategy, test_name):
+        total_summary = []
+        cs_popuplation_distribution = self.get_cs_distribution()
+        stress_cmd = stress_cmd_templ[0].replace("$DIST_PARAMS", cs_popuplation_distribution)
+
+        for current_ops in range(start_ops, max_ops + throttle_step, throttle_step):
+            self.log.info("Run cs command with rate: %s Kops", current_ops)
+            current_throttle = current_ops // (num_loaders * stress_num)
+            stress_cmd = stress_cmd.replace("$threads", f"{self.NUM_THREADS}").replace(
+                "$throttle", f"{current_throttle}")
+
+            stress_queue = self.run_stress_thread(stress_cmd=stress_cmd, stress_num=stress_num,
+                                                  compaction_strategy=compaction_strategy, stats_aggregate_cmds=False)
+            results = self.get_stress_results(queue=stress_queue, store_results=False)
+            self.log.info("All c-s results: %s", results)
+            summary_result = self._calculate_average_latency(results)
+            summary_result["ops"] = current_ops
+            self.log.info("C-S results for ops: %s. \n Results: \n %s", current_ops, summary_result)
+            total_summary.append(summary_result)
+            if summary_result["latency 95th percentile"] > self.MAX_95TH_LATENCY:
+                self.log.warning("Latency 95th percentile is large that %d", self.MAX_95TH_LATENCY)
+                break
+
+        total_summary = json.dumps(total_summary, indent=4, separators=(", ", ": "))
+        self.log.info("---------------------------------")
+        self.log.info("Final table with results: \n %s", total_summary)
+        self.log.info("---------------------------------")
+
+        filename = f"{self.logdir}/result_gradual_increase.log"
```

Comment:
will this file be collected during `collect logs` phase as well? if not, it could be a good idea to add it (if the size isn't too crazy)

Reply:
yes it will be collected.